pkg/controller/node/cidr_allocator.go

```diff
@@ -139,11 +163,19 @@ func (r *rangeAllocator) occupyCIDR(node *api.Node) error {
 // AllocateOrOccupyCIDR looks at the given node, assigns it a valid CIDR
 // if it doesn't currently have one or mark the CIDR as used if the node already have one.
 func (r *rangeAllocator) AllocateOrOccupyCIDR(node *api.Node) error {
+	if node == nil {
+		return nil
+	}
+	if !r.leaseNode(node.Name) {
```

Comment:
Please add a WARNING comment that anyone adding some "return" statements to this method should think about releasing the node.